### PR TITLE
[RAPTOR-3899] add anomaly inference template

### PR DESCRIPTION
**Pull request by jlpolit**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

### CODE REVIEW

**Comment by yakov-g** [comment]
We should a functional test as well.
And add ANOMALY target type into public API.

**Comment by yakov-g** [comment]
Could you add a new `anomaly` section somewhere here
https://github.com/datarobot/datarobot-user-models/blob/master/README.md#training-models

File: README.md [hunk 1]
```diff
@@ -267,6 +267,7 @@ The [model_templates](model_templates) folder contains sample models that work w
 
 ##### Inference Models
 * [Scikit-Learn sample model](model_templates/inference/python3_sklearn)
+* [Scikit-Learn sample unsupervised anomaly detection model](model_templates/inference/python3_sklearn_anomaly)
```
Comment (yakov-g) [inline README.md:270]: I thought as a different section:
- inference
- training
- anomaly

but up to you

**Review by yakov-g — APPROVED** [review]

**Comment by jlpolit** [comment]
> We should a functional test as well.
> And add ANOMALY target type into public API.

the latest master of the public api should work, but I don't think that code has made it into a published version yet- will definitely need to bump api version in requirements to get a working functional test for this

**Comment by yakov-g** [comment]
So does it make sense to add a test for this this template here:
https://github.com/datarobot/datarobot-user-models/blob/master/tests/functional/test_inference_model_templates.py
?

**Comment by jlpolit** [comment]
> So does it make sense to add a test for this this template here:
> https://github.com/datarobot/datarobot-user-models/blob/master/tests/functional/test_inference_model_templates.py
> ?

yeah, that's what I've done locally but it won't pass until we're using an updated dr public api client (since the last published version 2.22 doesn't account for anomaly target type)

**Review by meltzg — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (4dahalibut) [reply README.md:270]: I disagree

**Review by 4dahalibut — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (yakov-g) [reply README.md:270]: why

**Comment by yakov-g** [comment]
According to this: https://github.com/datarobot/datarobot-user-models/blob/master/jenkins/test3_drop_in_envs.sh
and `requirements_test.txt` datarobot==2.23.0b0 expected to be used

File: README.md (hunk 1 above)
Comment (4dahalibut) [reply README.md:270]: anomaly models aren't a third category, they fit the same way as multiclass within the inference training dichotomy.

**Comment by jlpolit** [comment]
> According to this: https://github.com/datarobot/datarobot-user-models/blob/master/jenkins/test3_drop_in_envs.sh
> and `requirements_test.txt` datarobot==2.23.0b0 expected to be used

interesting - seemingly that version doesn't match up with what I'm seeing on the public api master: https://github.com/datarobot/public_api_client/blob/master/datarobot/models/custom_model.py#L134  

not sure when that should be expected to be published/versioned

File: README.md (hunk 1 above)
Comment (yakov-g) [reply README.md:270]: you are right

**Comment by jlpolit** [comment]
will follow up with a functional test once this goes in https://github.com/datarobot/public_api_client/pull/1344

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**4.34** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**